pgcli/pgexecute.py

```diff
@@ -232,15 +232,22 @@ def connect(self, database=None, user=None, password=None, host=None,
             'dsn': dsn,
         }
         new_params.update(kwargs)
+
+        if 'dsn' in new_params:
+            new_params = {
+                'dsn': new_params['dsn'],
+                'password': new_params['password']
+            }
+
+            if 'password' in new_params:
```

Comment:
Same here: you only want to add `password` into dsn string if password is non-empty.